File: PanteonGame.UI/src/helpers/models/Building.ts
import { IBaseParam, IBaseResult } from "./BaseModel";

export interface IBuildingParam extends IBaseParam{
    buildingType: string;
    buildingCost?: number;
    constructionTime?: number;
    isDeleted? : boolean;
}
export interface IBuildingResult extends IBaseResult { 
    buildingType: string;
    buildingCost: number;
    constructionTime: number;
    isDeleted?: boolean
}

File: PanteonGame/Entities/Building/Building.cs
using Entities.MongoDbEntity;

namespace Entities.Building
{
    public class Building : BaseMongoDbEntity
    {
        public string BuildingType { get; set; }
        public double BuildingCost { get; set; }
        public int ConstructionTime { get; set; }
        public bool IsDeleted { get; set; }

    }
}
